staging_service/autodetect/GenerateMappings.py

```diff
@@ -0,0 +1,262 @@
+        "output_type": ["KBaseBiochem.Media"],
+    },
+    # {
+    #     "title": "Attribute Mapping",
+    #     "app": "kb_uploadmethods/import_attribute_mapping_from_staging",
+    #     "output_type": ["KBaseExperiments.AttributeMapping"],
+    # },
+    {
+        "title": "Expression Matrix",
+        "app": "kb_uploadmethods/import_tsv_as_expression_matrix_from_staging",
+        "output_type": ["KBaseFeatureValues.ExpressionMatrix"],
+    },
+    {
+        "title": "Metabolic Annotations",
+        "app": "MergeMetabolicAnnotations/import_annotations",
+        "output_type": ["KBaseGenomes.Genome"],
+    },
+    {
+        "title": "Bulk Metabolic Annotations",
+        "app": "MergeMetabolicAnnotations/import_bulk_annotations",
+        "output_type": ["KBaseGenomes.Genome"],
+    },
+    {
+        "title": "FBA Model",
+        "app": "kb_uploadmethods/import_file_as_fba_model_from_staging",
+        "output_type": ["KBaseFBA.FBAModel"],
+    },
+    {
+        "title": "Phenotype Set",
+        "app": "kb_uploadmethods/import_tsv_as_phenotype_set_from_staging",
+        "output_type": ["KBasePhenotypes.PhenotypeSet"],
+    },
+]
+
+mapping[EXCEL] = [
+    {
+        "title": "Samples",
+        "app": "sample_uploader/import_samples",
+        "output_type": ["KBaseSets.SampleSet"],
+    },
+    {
+        "title": "Media",
+        "app": "kb_uploadmethods/import_tsv_excel_as_media_from_staging",
+        "output_type": ["KBaseBiochem.Media"],
+    },
+    {
+        "title": "Attribute Mapping",
+        "app": "kb_uploadmethods/import_attribute_mapping_from_staging",
+        "output_type": ["KBaseExperiments.AttributeMapping"],
+    },
+    {
+        "title": "FBA Model",
+        "app": "kb_uploadmethods/import_file_as_fba_model_from_staging",
+        "output_type": ["KBaseFBA.FBAModel"],
+    },
+]
+mapping[JSON] = [
+    {
+        "title": "EscherMap",
+        "app": "kb_uploadmethods/import_eschermap_from_staging",
+        "output_type": ["KBaseFBA.EscherMap"],
+    }
+]
+
+mapping[SBML] = [
+    {
+        "title": "FBA Model",
+        "app": "kb_uploadmethods/import_file_as_fba_model_from_staging",
+        "output_type": ["KBaseFBA.FBAModel"],
+    }
+]
+
+new_apps = OrderedDict()
+extensions_flat = []
+
+counter = 0
+for category in mapping:
+    apps = mapping[category]
+    for app in apps:
+        # print("looking at", app)
+        title = app["title"]
+        if title not in new_apps:
+            app["extensions"] = []
+            app["id"] = counter
+            counter += 1
+            new_apps[title] = copy.copy(app)
+
+        for extension in type_to_extension_mapping:
+            if extension == category:
+                new_apps[title]["extensions"].extend(
+                    type_to_extension_mapping[extension]
+                )
+                extensions_flat.extend(type_to_extension_mapping[extension])
+
+
+extensions_mapping = defaultdict(list)
+for app in new_apps:
```

Comment:
same as above, would be nice to have a brief comment explaining what's happening here